src/ssh/sftp_dir_iterator.cpp

```diff
@@ -70,8 +69,8 @@ SFTPAttributesUPtr SFTPDirIterator::next()
         return next();
     }
 
-    auto err_msg = fmt::format("[sftp] cannot read remote directory '{}': {}", dir->name, ssh_get_error(sftp->session));
+    SFTPError err{"cannot read remote directory '{}': {}", dir->name, ssh_get_error(sftp->session)};
     dirs.pop();
-    throw std::runtime_error{err_msg};
+    throw std::move(err);
```

Comment:
Better remove the move here. The reasoning is similar to return statements: the standard already requires that err be treated as an rvalue in this case and, by using the cast, you prevent the compiler from entirely eliding the move because the operand of the throw exception is no longer an automatic local object.